CMakeLists.txt

```
@@ -75,27 +75,64 @@ option(DPNP_USE_ONEMKL_INTERFACES
     "Build DPNP with oneMKL Interfaces"
     OFF
 )
+set(HIP_TARGETS "" CACHE STRING "HIP architecture for target")
```

I assume there is no support for multiple values:
```suggestion
set(HIP_TARGET "" CACHE STRING "HIP architecture for target")
```

---

At some point, it was clear in docs that only one architecture was supported at a time, but now it isn't as clear and should be tested

Also, there is new information in the [extension guide](https://developer.codeplay.com/products/oneapi/amd/2025.1.0/guides/get-started-guide-amd#compile-for-multiple-targets)

```
The compiler driver also offers alias targets for each target+architecture pair to make the command line shorter and easier to understand for humans. Thanks to the aliases, the -Xsycl-target-backend flags no longer need to be specified.
```

It shows that the command
```
icpx -fsycl -fsycl-targets=spir64_gen,amdgcn-amd-amdhsa,nvptx64-nvidia-cuda \
        -Xsycl-target-backend=spir64_gen '-device pvc' \
        -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx1030 \
        -Xsycl-target-backend=nvptx64-nvidia-cuda --offload-arch=sm_80 \
        -o sycl-app sycl-app.cpp
```
is equivalent to
```
icpx -fsycl -fsycl-targets=intel_gpu_pvc,amd_gpu_gfx1030,nvidia_gpu_sm_80 \
        -o sycl-app sycl-app.cpp
```
so maybe both dpctl and dpnp can simplify by removing the need for `-Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=[X]` completely

list of aliases:
https://intel.github.io/llvm/UsersManual.html

---

Aliases list seems to claim only one alias is supported at a time. So probably only one architecture at once is possible? That would be my guess

---

> I assume there is no support for multiple values:

I am using `HIP_TARGETS` instead of `HIP_TARGET` because oneMath requires `HIP_TARGETS` to be defined
https://github.com/uxlfoundation/oneMath/blob/4ad4dfb5db834117248ad5f8fbded5cfc1097005/CMakeLists.txt#L73

---

Is it only declaration of cmake variable which doesn't impact oneMath, isn't that?

---

> Is it only declaration of cmake variable which doesn't impact oneMath, isn't that?

According to [OneMath documentation](https://github.com/uxlfoundation/oneMath/blob/4ad4dfb5db834117248ad5f8fbded5cfc1097005/docs/building_the_project_with_dpcpp.rst#building-for-rocm) HIP_TARGETS must be set for ROCm builds

---

> At some point, it was clear in docs that only one architecture was supported at a time, but now it isn't as clear and should be tested
> 
> Also, there is new information in the [extension guide](https://developer.codeplay.com/products/oneapi/amd/2025.1.0/guides/get-started-guide-amd#compile-for-multiple-targets)
> 
> > The compiler driver also offers alias targets for each target+architecture pair to make the command line shorter and easier to understand for humans. Thanks to the aliases, the -Xsycl-target-backend flags no longer need to be specified.
> 
> It shows that the command
> 
> ```
> icpx -fsycl -fsycl-targets=spir64_gen,amdgcn-amd-amdhsa,nvptx64-nvidia-cuda \
>         -Xsycl-target-backend=spir64_gen '-device pvc' \
>         -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx1030 \
>         -Xsycl-target-backend=nvptx64-nvidia-cuda --offload-arch=sm_80 \
>         -o sycl-app sycl-app.cpp
> ```
> 
> is equivalent to
> 
> ```
> icpx -fsycl -fsycl-targets=intel_gpu_pvc,amd_gpu_gfx1030,nvidia_gpu_sm_80 \
>         -o sycl-app sycl-app.cpp
> ```
> 
> so maybe both dpctl and dpnp can simplify by removing the need for `-Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=[X]` completely
> 
> list of aliases: https://intel.github.io/llvm/UsersManual.html

This is a great suggestion.
1. The compiler supports more than one target
2. Using aliases greatly simplifies the logic

In this PR I will implement using aliases for AMD,
In the next PR I will suggest an update for CUDA